Trouble authenticating against Windows 2012r2

Here's my test client that I'm trying to get working against my Windows 2012r2 AD environment which I figure should be the bare minimum that should  work:
```golang
package main

import "gopkg.in/jcmturner/gokrb5.v2/config"
import "gopkg.in/jcmturner/gokrb5.v2/client"
//import "gopkg.in/jcmturner/gokrb5.v2/keytab"
import "log"
import "os"

func main() {
        l := log.New(os.Stderr, "GOKRB5 Client: ", log.Ldate|log.Ltime|log.Lshortfile)

        cfg, err := config.Load("/etc/krb5.conf")
        if err != nil {
                l.Printf("Error on config: %v\n", err)
        }

        //kt, err := keytab.Load("/home/user/nsupdate.keytab")
        //if err != nil {
        //        l.Printf("Error on keytab: %v\n", err)
        //}

        cl := client.NewClientWithPassword("user", "EXAMPLE.COM", "password")
        cl.WithConfig(cfg)
        err = cl.Login()
        if err != nil {
                l.Printf("Error on AS_REQ: %v\n", err)
        }
}
```
With this I get the error ` Error on AS_REQ: [Root cause: Encrypting_Error] KRBMessage_Handling_Error: AS Exchange Error: failed setting AS_REQ PAData for pre-authentication required < Encrypting_Error: Error creating etype: [Unknown or unsupported EType: 26]`. Looking up EType 26 suggests it's one of the Camellia ciphers `camellia256-cts-cmac` (defined in RFC 6803), I'm a bit suprised as all my searching suggests Windows doesn't support that.

I've found if I edit the user in AD and tick the option `Do not require Kerberos preauthentication` then I get this different error `Error on AS_REQ: [Root cause: KRBMessage_Handling_Error] KRBMessage_Handling_Error: AS Exchange Error: AS_REP is not valid < KRBMessage_Handling_Error: KDC did not respond appropriately to FAST negotiation`. Now I'm a bit stuck. I'd like to eventually use a keytab but for now I'm just trying to get it passing with a password.

Ultimately, I'm trying to work out if it's possible to use your library along with `github.com/miekg/dns` to be able to perform GSS-TSIG DNS updates to a Windows DNS/AD server as they don't support normal key-based TSIG updates like BIND does or otherwise require you to turn off secure DNS updates altogether, which won't be an option. I can currently do this manually with a combination of `ktutil`, `kinit` and `nsupdate -g` but I would like a Go equivalent.

You might be able to tell me straight off if your library has enough of the GSSAPI implemented for me to even do that.

If I don't also prune `rc4-hmac` from that list, I get this error:
```
GOKRB5 Client: 2017/11/27 15:29:39 test.go:29: Error on AS_REQ: [Root cause: Encrypting_Error] KRBMessage_Handling_Error: AS Exchange Error: failed setting AS_REQ PAData for pre-authentication required < Encrypting_Error: Error getting key from credentials: [Error deriving key from string: Invalid s2kparams length]
```
Other than that, my test client now logs in successfully provided I add the `cl.GoKrb5Conf.DisablePAFXFast = true` as you indicated.

--- a/config/krb5conf.go
+++ b/config/krb5conf.go
@@ -544,7 +544,7 @@ func parseETypes(s []string, w bool) []int {
 				continue
 			}
 		}
-		i := etypeID.ETypesByName[et]
+		i := etypeID.EtypeSupported(et)
 		if i != 0 {
 			eti = append(eti, i)
 		}

--- a/iana/etypeID/constants.go
+++ b/iana/etypeID/constants.go
@@ -75,3 +75,27 @@ var ETypesByName = map[string]int{
 	"camellia256-cts":              CAMELLIA256_CTS_CMAC,
 	"subkey-keymaterial":           SUBKEY_KEYMATERIAL,
 }
+
+// EtypeSupported resolves the etype name string to the etype ID.
+// If zero is returned the etype is not supported by gokrb5.
+func EtypeSupported(etype string) int {
+	// Slice of supported enctype IDs
+	s := []int{
+		AES128_CTS_HMAC_SHA1_96,
+		AES256_CTS_HMAC_SHA1_96,
+		AES128_CTS_HMAC_SHA256_128,
+		AES256_CTS_HMAC_SHA384_192,
+		DES3_CBC_SHA1_KD,
+		RC4_HMAC,
+	}
+	id := ETypesByName[etype]
+	if id == 0 {
+		return id
+	}
+	for _, sid := range s {
+		if id == sid {
+			return id
+		}
+	}
+	return 0
+}

